R/caption.R

```diff
@@ -0,0 +1,153 @@
+#' More sophisticated parsing of caption
+#'
+#' @noRd
+extract_short <- function(x) {
+  x <- trimws(x)
+  open <- gregexpr("[", x, fixed = TRUE)[[1]]
+  close <- gregexpr("]", x, fixed = TRUE)[[1]]
+  mismatch <- length(open) != length(close)
+  positions <- c(open, close)
+  if(length(open)==0 || length(close)==0 || open[1] != 1) {
+    return(list(main = x, short = ""))
+  }
+  bits <- c(rep(1, length(open)), rep(-1, length(close)))
+  o <- order(positions)
+  bits <- bits[o]
+  positions <- positions[o]
+  done <- cumsum(bits)
+  done <- positions[done==0]
+  if(length(done)==0) {
+    if(mismatch) {
+      done <- positions[bits==-1][1]
+    } else {
+      return(list(main = x, short = ""))
```

Comment:
The tests don't hit this line, and I couldn't come up with a case where `done` would have no 0 elements but brackets were matched.  Am I missing one?

(If it should be inaccessible given upstream logic, signaling an error would make things clearer in my opinion, though I think it's okay as is.)

Reply:
I think this might be handled now by this bit upstream. I'll drop this whole block as unneeded now. 
```r
if(length(open)==0 || length(close)==0 || open[1] != 1) {
    return(list(main = x, short = ""))
  }
```

Reply:
Ok ... revisiting this. Right ... so basically if `done` is never 0, then I'll just take the last close bracket and leave it at that. I think that's what you're addressing, right? The check of `mismatch` doesn't matter.

Reply:
> I think that's what you're addressing, right?

Yes, that takes care of what I was talking about (the unreachable `else` arm).